### PR TITLE
Add setInterval to YouTube watch page

### LINKED ISSUE #119: Min Vid icon is missing from YouTube video
[Affected versions]:
Firefox 45 and up
Latest Min Vid version (built from GitHub repo)

[Affected Platforms]:
All Windows
All Mac OS X
All Linux

[Prerequisites]:
Latest Min Vid add-on was previously installed on a profile.

[Steps to reproduce]:
1. Open the browser with the profile from prerequisites.
2. Go to www.youtube.com .
3. Click on any video available.
4. Observe the Min Vid icon that is displayed on top of the video.

[Expected result]:
The Min Vid icon is overlaid in the upper right corner of the video.

[Actual result]:
The Min Vid icon is missing.

[Notes]:
1. If another video is opened in a new tab using the context menu or the middle mouse button, then the Min Vid icon is displayed.
2. [Here](https://drive.google.com/open?id=0B6QogV2AJFKieUpwWkhEZDVRUFU) is a screen recording describing the issue

### PR DESCRIPTION
**Pull request by kaganjd**
Fixes #119.

### CODE REVIEW

File: data/icon-overlay.js [hunk 1]
```diff
@@ -23,19 +23,21 @@ if ((host.indexOf('youtube.com') > -1) && ytHomeContainers.length) {
...
-const ytWatchContainer = document.querySelector('.html5-video-player');
-if ((host.indexOf('youtube.com') > -1) && ytWatchContainer) {
-  ytWatchContainer.classList.add('minvid__overlay__wrapper');
-  const tmp = getTemplate();
-  tmp.addEventListener('click', function(ev) {
-    evNoop(ev);
-    self.port.emit('launch', {
-      url: window.location.href,
-      domain: host
+setTimeout(() => {
```
Comment (meandavejustice) [inline data/icon-overlay.js]: This should be `setInterval`

Comment (kaganjd) [inline data/icon-overlay.js]: Doh, my bad. Updated.

**Comment by meandavejustice** [comment]
It looks like what you currently have will add the icon every two seconds, we want to just check that nothing has changed, and add an icon if it is not visible on an interval.

File: data/icon-overlay.js [hunk 2]
```diff
@@ -23,19 +23,21 @@ if ((host.indexOf('youtube.com') > -1) && ytHomeContainers.length) {
...
-const ytWatchContainer = document.querySelector('.html5-video-player');
-if ((host.indexOf('youtube.com') > -1) && ytWatchContainer) {
-  ytWatchContainer.classList.add('minvid__overlay__wrapper');
-  const tmp = getTemplate();
-  tmp.addEventListener('click', function(ev) {
-    evNoop(ev);
-    self.port.emit('launch', {
-      url: window.location.href,
-      domain: host
+setInterval(() => {
```
Comment (6a68) [inline data/icon-overlay.js]: The first thing to do inside a `setInterval` callback is to call `clearTimeout` to clear it. You'll need to assign the interval ID (returned by `setInterval`), something like:

```
let ytWatchContainerInterval = setInterval(() => { 
  clearInterval(ytWatchContainerInterval);

  // then back to regularly scheduled programming:
  const ytWatchContainer = document.querySelector('.html5-video-player');
  // ...
```

Comment (6a68) [inline data/icon-overlay.js]: Oops, I'm wrong. You want to cancel the interval as soon as you find the element you're after

Comment (6a68) [inline data/icon-overlay.js]: @kaganjd So, you might rearrange your patch to do something like:

- if you are on youtube.com, start the interval.
- inside the callback, if ytWatchContainer is truthy (querySelector will return null if the element isn't found), clear the interval, then do the rest of the stuff starting with adding the overlay__wrapper class

**Comment by meandavejustice** [comment]
closing in favor of https://github.com/meandavejustice/min-vid/pull/170, trying to get this mvp milestone done before user testing next week, sorry @kaganjd!!

**Pull request closed without merging**